technical-doc/storage-costs.rst

```diff
@@ -29,5 +29,13 @@ To create your address on the blockchain someone need to send at least 0.00025 *
 
 You want to store your birth date that is 30 bytes long in your datastore so you need to send an operation that will create a key in your datastore using a SC. This operation will cost you in storage costs at least 0.00025 * (10 + 30) = 0.01 Massa.
 
-Now you want to delete this entry on your datastore then you will do it with a SC and you will be refunded of the storage costs (0.02 Massa).
+Now you want to delete this entry on your datastore you will be refunded of the storage costs (0.02 Massa).
 
+In case you are interacting with a SC that will store the data on his datastore then he will be charged for the storage costs not you.
+
+Notes
+-----
+
+In case of refund of the storage costs, after realeasing space, the address reimbursed is the one that calls the ABI, possibly different from the one that paid for this storage.
+
+If you are a SC developer and you want your users to pay for the storage costs of your smart contract you can use the coins that are passed by the `coins` parameter of CallSC.
```

Comment:
This would imply that the contract needs to check how many coins the user sent, and if there are enough to execute the remaining instructions of the function then proceed to execution or revert the transaction, wouldn't it ? Otherwise the SC would just use its balance to execute the remaining instructions even the user didnt send enough coins, no ?

Reply:
It's possible that the SC check his balance as start of the code of his function and check it at the end with an assert and so if it fails then the execution is automatically revert by the core. We plan to add the number of coins used in the execution in read only call so that fronts of dApps can simulate and then fill the `coins` parameter the right way